.github/scripts/boost/build_and_test_boost.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+BOOST_BUILD_DIR=$1
+BOOST_TEMPLATE_DIR=$2
+
+if [[ $# -lt 1 ]]; then
+    echo "Expected at least 1 arguments!" >&2
+    exit 1
+fi
+
+if [[ -d $BOOST_BUILD_DIR ]]; then
+    echo "Directory $BOOST_BUILD_DIR already exists!" >&2
+    exit 1
+fi
+
+mkdir -p $BOOST_BUILD_DIR
+
+if [[ ! -z $BOOST_TEMPLATE_DIR && -d $BOOST_TEMPLATE_DIR ]]; then
+    echo "Found Boost template dir '$BOOST_TEMPLATE_DIR', using it instead of cloning.."
+    time cp -r $BOOST_TEMPLATE_DIR/* $BOOST_BUILD_DIR
+else
+    echo "Cloning Boost GitHub repository.."
+    time git clone --recursive https://github.com/boostorg/boost.git $BOOST_BUILD_DIR
```

Comment:
Currently, we have a specific script for toolchain sources cloning (`update-sources.sh`) but nothing like that for individual tested packages and their sources must be either clonned manually or in the GitHub Actions workflow. This would deserve some refactoring but for now, moving the clonning of Boost to a separate script would be enough.